### ISSUE TITLE
pytest_xdist breaks tests for bash-completion on armv7l

### ISSUE THREAD

**Issue by Patagonicus** [NixOS/nixpkgs#91706]
**Describe the bug**
Trying to install the bash-completion package on armv7l fails due to multiple test failures. This seems to be caused by a85b07cbcb7a034bc07dda3642bc68fe621a63ec as reverting that commit makes the tests pass.

Build log at commit 22a81aa5fc15b2d41b12f7160a71cd4a9f3c3fa1 (nixos-unstable when I was testing): http://sprunge.us/Q9an7c
Build log at the same commit after `git revert a85b07cbcb7a034bc07dda3642bc68fe621a63ec`: http://sprunge.us/lJTzhY

**To Reproduce**
`nix-shell -p bash-completion`

**Expected behavior**
Build succeeds.

**Screenshots**
N/A

**Additional context**
I'm running this on an Odroid HC2 machine, so native armv7l, not armv7l on AArch64 or something like that.

Without xdist the tests take considerably longer to run. Maybe using xdist can be made conditional on the architecture?

**Notify maintainers**

@andir  who introduced the usage of xdist for the tests
@peti @xfix

**Metadata**
 - system: `"armv7l-linux"`
 - host os: `Linux 5.4.33, NixOS, 20.09pre-git (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos-20.09pre231796.22a81aa5fc1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: bash-completion
# a list of nixos modules affected by the problem
module:
```

**Comment by Patagonicus** [comment]
`thefloweringhash` on IRC pointed out to me that the 8 cores of the HC2 are more than many other armv7l machines, so I ran the tests with `-n 2` instead of `-n $NIX_BUILD_CORES`. That also fails, but it looks like it's less tests timing out: http://sprunge.us/ZMC3Mo

**Comment by Patagonicus** [comment]
I did another run. It's not deterministic. The second try with `-n 2` passed.

The original commit already mentions that the tests are flaky. I'm not sure that tests failing gives any meaningful data, at least when the tests are run in parallel.

**Comment by lopsided98** [comment]
I'm seeing this as well on the same hardware. I wrote a patch to increase the timeout, but it still fails even if I increase it to 500 seconds if my machine is running other builds. The tests that fail seem pretty much random. If my machine has no load, it succeeds even with the original (30 second) timeout.

**Comment by Patagonicus** [comment]
I've done some more testing, this time on an x86_64 machine. It's an Intel NUC with an i3-7100U. This is a 2 core/4 thread processor.

While the tests usually pass on an unloaded machine, even watching a Youtube video in Chrome at the same time made them fail for me. I also tried with `stress -c 16` to simulate a heavily loaded system. With the tests running in parallel they failed more than half the time out of a few runs I did, but having them run serially I couldn't get them to fail at all.

I'm sending a pull request to revert the cell that made the tests parallel since I'm a pretty firm believer that flaky tests are worse than no tests.
